app/scripts/directives/editEnvironmentFrom.js

```diff
@@ -50,11 +50,12 @@
       ctrl.editEnvironmentFromForm.$setDirty();
     };
 
+    ctrl.hasOptions = function() {
+      return !_.isEmpty(ctrl.envFromSelectorOptions);
+    };
+
     ctrl.isEnvFromReadonly = function(entry) {
-      return ctrl.isReadonlyAny ||
-        entry.isReadonlyValue === true ||
-        ((entry.secretRef || entry.configMapRef) && !entry.selectedEnvFrom) ||
-        _.isEmpty(ctrl.envFromSelectorOptions);
+      return entry.isReadonlyValue === true || ((entry.secretRef || entry.configMapRef) && !entry.selectedEnvFrom);
```

Comment:
I was proposing getting rid of `isReadonlyValue` and keeping only `isReadonlyAny` :)

I don't think we have a need for only certain values being read only. Either the entire list is read only or it isn't

Reply:
Hahahaha whoops, I'll go the other direction

Reply:
After trying to pick this apart looks like the piece around Secrets or Config maps and duplicate entries falls into a strange area. Specifically around when a user edits the Yaml and purposefully enters duplicates. 

With the current version these duplicates become read only because they lack the additional property around "selectedEnvFrom" due to the way lodash and find is being used... this is unintended but has kinda given us an inadvertent solution to the duplicate issue.

Reply:
I'm not sure I understand why we made duplicates read only, or missing secrets for that matter. Why prevent the  user from editing those to fix the problem?

I think it should be a warning if it's there twice, but we shouldn't block it since it's not a validation error.

Reply:
Concur on the blocking at all, that shouldn't happen, and doesn't... kinda. The current implementation doesn't block a user from deleting or rearranging an item, just creating an additional duplicate/triplicate/quad/etc entry through the interface. 

Can and will look at using lodash filter instead of find to handle the checks, this may correct the problem, keep you posted

Reply:
@spadgett looks like the aiming towards ```_.filter``` achieved the desired result, no more entries being read only if they're duplicates. Leveraged the same concept to apply a temporary placeholder ```duplicate``` class around the individual UI-Selects

@spadgett  & @benjaminapetersen 

On the Readonly state as an all or nothing piece... can push them up, just need a confirmation for the conditions on my local, which so far encompass... 
- If options aren't populated, everything is Read Only
- If either the Secrets OR Config Maps "Can I" checks fail, everything is Read Only
- And finally, removed the block (similar to below) for binding attribute pass-through for Read Only 
  ```js
    if('isReadonly' in $attrs) {
      ctrl.isReadonlyAny = true;
    }
  ```